service/payment.go

```diff
@@ -258,6 +259,20 @@ func getCosts(resource string, cfg *config.Config) (*models.CostsRest, ResponseT
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode == http.StatusNotFound {
+		r := regexp.MustCompile(cfg.SecureAppCostsRegex)
```

Comment:
rather than compiling every time this case is encountered this compile function should be run on startup, easiest place seems to be on creation of the payment service struct here https://github.com/companieshouse/payments.api.ch.gov.uk/blob/7f601d95b3f329375890ccafc505185bc3a9dc7a/handlers/register.go#L24

since this is then run on startup it will be clear whether the regex can be parsed or not (MustCompile will panic rather than return an error if it doesn't so the service won't run)

when moving this into the handler registration I would also switch back to Compile rather than MustCompile to make it simpler to determine any issues by handling the error explicitly then ending the process with an "os.Exit(1)" after logging the error